lib/workspace.rb

```diff
@@ -0,0 +1,51 @@
+module Slack
+  class Workspace
+    attr_reader :users, :channels, :selected
+
+    def initialize
+      @users = User.list
+      @channels = Channel.list
+      @selected = nil
+    end
+
+    def select_channel(input)
+      select(input, channels)
+    end
+
+    def select_user(input)
+      select(input, users)
+    end
+
+    def send_message(message)
+      selected.send_message(message)
+    end
+
+    def tp_details_options
```

Comment:
Instead of doing this in the Controller/Workspace, you could instead just return `selected.details` if you'd implemented that method in the `User` or `Channel` classes.